runtime/lua/vim/_meta/misc.lua

```diff
@@ -14,3 +14,23 @@
 --- @param ... any
 --- @return any
 function vim.call(func, ...) end
+
+--- @class vim.context.mods
+--- @field sandbox? boolean
+--- @field noautocmd? boolean
+--- @field hide? boolean
+--- @field horizontal? boolean
+--- @field keepalt? boolean
+--- @field keepjumps? boolean
+--- @field keepmarks? boolean
+--- @field keeppatterns? boolean
+--- @field lockmarks? boolean
+---
+--- @field buf? integer
+--- @field win? integer
+
+--- @generic R1, R2, R3, R4
+--- @param mods vim.context.mods
+--- @param f fun(): R1, R2, R3, R4
+--- @return R1, R2, R3, R4
+function vim._context(mods, f) end
```

Comment:
Extra benefit: this connects all of these scattered concepts into a single "handle" that can be referred to in documentation and helps users build a more powerful mental model of the system.